docs/topics/compatibility-guides/compatibility-guide-21.md

```diff
@@ -0,0 +1,479 @@
+>
+> **Short summary**: K2 changes the behavior of smart cast propagation, introducing bidirectional propagation of type
+> information for inferred variables, like `val x = y`. Explicitly typed variables, such as `val x: T = y`,
+> no longer propagate type information, ensuring stricter adherence to declared types.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: enable the new behavior
+
+### Correct the handling of member-extension property overrides in Java subclasses
+
+> **Issue**: [KTLC-35](https://youtrack.jetbrains.com/issue/KTLC-35)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: The getter for member-extension properties overridden by Java subclasses is now hidden in the subclass's scope,
+> aligning its behavior with regular Kotlin properties.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: enable the new behavior
+
+### Correct visibility alignment for getters and setters of var properties overriding a protected val
+
+> **Issue**: [KTLC-36](https://youtrack.jetbrains.com/issue/KTLC-36)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: binary
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: enforce consistent visibility for both getters and setters in K2; K1 remains unaffected
+
+### Raise severity of JSpecify nullability mismatch diagnostics to errors
+
+> **Issue**: [KTLC-11](https://youtrack.jetbrains.com/issue/KTLC-11)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Nullability mismatches from `org.jspecify.annotations`, such as `@NonNull`, `@Nullable`, and `@NullMarked` are treated as errors instead of warnings,
+> enforcing stricter type safety for Java interoperability. To adjust the severity of these diagnostics, use the `-Xnullability-annotations` compiler option.
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: report a warning for potential nullability mismatches
+> - 1.8.20: expand warnings to specific JSpecify annotations, including: `@Nullable`, `@NullnessUnspecified`, `@NullMarked`, and legacy annotations in `org.jspecify.nullness` (JSpecify 0.2 and earlier)
+> - 2.0.0: add support for the `@NonNull` annotation
+> - 2.1.0: change default mode to strict  for JSpecify annotations, converting warnings into errors; use `-Xnullability-annotations=@org.jspecify.annotations:warning` or `-Xnullability-annotations=@org.jspecify.annotations:ignore` to override the default behavior
+
+### Change overload resolution to prioritize extension functions over invoke calls in ambiguous cases
+
+> **Issue**: [KTLC-37](https://youtrack.jetbrains.com/issue/KTLC-37)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: change overload resolution to consistently prioritize extension functions over `invoke` calls for extension functions with matching signatures; this change applies only after recompilation and does not affect precompiled binaries
```

Comment:
Actually, this sounds like the summary. Maybe you can take details from here and leave this explanation simpler.